content/blog/2022/2022-09-25-node-urlShortener.md

```diff
@@ -0,0 +1,531 @@
+- Click on Deploy a shared cloud database for `Free`
+- Click on Create a Shared Cluster
+- Click on **Database access** on the sidebar and Add New Database User
+- Select Password then enter in a username and password detail for your user.
+- `Built-in Role` select `Atlas Admin`
+- Click the **Add User** button to create your new user
+- Click on **Network Access** on the sidebar.
+To Allow Access From All IP Addresses
+- Click on Add IP Address button
+- Select `ALLOW ACCESS FROM ANYWHERE`
+- Click the **Confirm**  button.
+- Click on **Database** on the sidebar
+- Click the  **Connect**  button for your cluster
+- In the popup modal, click on  **Connect your application**.
+- **Copy** the URI on the clipboard
+- Lastly, all you need to do is replace the `<password>` field with the password you created previously.
+
+Our MongoDB atlas is all set and ready for use. Otherwise, click [here](https://www.mongodb.com/basics/mongodb-atlas-tutorial#:~:text=Getting%20Started%20with%20MongoDB%20Atlas&text=Create%20a%20MongoDB%20Cloud%20account,Connect%20to%20the%20cluster.) for a more in-depth guide on how to set up a MongoDB cluster.
+
+To secure and keep our MongoDB Atlas URI confidential, we will store the copied URI link in the `.env` file we created above.
+
+Copy and Paste the following code into our `.env` file:
+```env
+MONGO_URI= mongodb+srv://<username>:<password>@cluster0.oq1hdin.mongodb.net/?retryWrites=true&w=majority
+DOMAIN_URL=http://localhost:3333
+```
+Here, we are storing  `MongoDB_URI` and `DOMAIN_URL` as environment variables in the `.env` file. Replace the `MongoDB_URI` link with the one you generated in MongoDB Altas and remember to input your `username` and `password`. While `DOMAIN_URL` is our server's `localhost` address, which can readily be changed during production.
+
+To query and connect to our MongoDB database easily, we will use the Object Data Modeling (ODM) library **Mongoose**.
+
+Mongoose manages relationships between data, provides schema validation, and is used to translate between objects in code and the representation of those objects in MongoDB. it remains one of the most popular ODM tools for MongoDB. If you are coming from a SQL background then using Mongoose will make the transition into a NoSQL environment much easier. It will also save you time writing your own validations and instance methods and is highly recommended for basic MongoDB operations.
+
+While MongoDB is schema-less, SQL defines a schema via the table definition. Mongoose provides us with schema validation a document data structure (shape of the document) that is enforced via the application layer.
+
+Mongoose schema defines the structure of the document, default values, validators, etc.
+
+## Creating MongoDB Schema
+To use mongoose we have to start by creating schemas for our application. Each schema maps to a MongoDB collection and defines the shape of the documents within that collection. We will need to create a Schema for our URLs. 
```

Comment:
The first sentence in the previous section says that MongoDB is schemaless. Here you write that we HAVE to start by creating a schema. Can you modify this to make it sound less contradicting? :)